streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/promise/exception/CancelledPromiseException.kt

```diff
@@ -0,0 +1,3 @@
+package com.mirego.trikot.streams.reactive.promise.exception
+
+object CancelledPromiseException : Exception()
```

Comment:
Est-ce que tu voudrais que tes exceptions de Promise soitent dans une Sealed classes ?

Reply:
Something like the following?

```kotlin
sealed class PromiseException
object CancelledPromiseException: PromiseException()
object EmptyPromiseException: PromiseException()
```

Reply:
Yes, [the key benefit of using sealed classes comes into play when you use them in a when expression. If it's possible to verify that the statement covers all cases, you don't need to add an else clause to the statement. However, this works only if you use when as an expression (using the result) and not as a statement.](https://kotlinlang.org/docs/reference/sealed-classes.html)

Reply:
I think you can set 
`sealed class PromiseException : Exception()` if you want to keep it as an exception

Reply:
Got it! Thanks for the tips 🙂